lib/cli-engine.js

```diff
@@ -391,12 +392,14 @@ function getCacheFile(cacheFile, cwd) {
  */
 function CLIEngine(options) {
 
+    options = (options instanceof Object) ? assign(options) : {};
+    options.cwd = options.cwd || process.cwd();
+
     /**
      * Stored options for this instance
      * @type {Object}
      */
-    this.options = assign(Object.create(defaultOptions), options || {});
-
+    this.options = assign(defaultOptions, options);
```

Comment:
By doing this a lot of default values will be overwritten by undefined.
Also you can remove line 395-396 and revert this change, you will get what you are looking for here. Basically you don need these changes.